### PR TITLE
Fix deprecated costmap params

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: cob_navigation_config/cob3-2/costmap_common_params.yaml [hunk 1]
```diff
@@ -1,30 +1,97 @@
...
 
-laser_scan_rear: {sensor_frame: base_laser_rear_link, topic: /base_laser_rear/scan, data_type: LaserScan, marking: true, clearing: true}
+#
+# Configure the inflation_layer
+#
+inflation_layer:
 
-laser_scan_top: {sensor_frame: base_laser_top_link, topic: /base_laser_top/scan, data_type: LaserScan, marking: true, clearing: true}
+  # The radius in meters to which the map inflates obstacle cost values.
+  inflation_radius: 0.55 # default: 0.55
```
Comment (ipa-flg) [inline cob_navigation_config/cob3-2/costmap_common_params.yaml:94]: I know that value for inflation_radius is default, but makes no sense with the robot's circumscribed radius already larger than 0.55m. It should be inscribed_radius <= circumscribed_radius <= inflation_radius 
Also cost_scaling_factor is not properly chosen, a value of 8.01 rather suggests an inflation_radius of about 1.0m.

@ipa-mig as those are default values, I'd suggest to keep the values that way for this PR. But we should discuss this, and make a separate PR for updating these values when indicated.

File: cob_navigation_config/cob3-6/costmap_common_params.yaml [hunk 2]
```diff
@@ -1,30 +1,97 @@
...
 
-laser_scan_rear: {sensor_frame: base_laser_rear_link, topic: /base_laser_rear/scan, data_type: LaserScan, marking: true, clearing: true}
+#
+# Configure the inflation_layer
+#
+inflation_layer:
 
-laser_scan_top: {sensor_frame: base_laser_top_link, topic: /base_laser_top/scan, data_type: LaserScan, marking: true, clearing: true}
+  # The radius in meters to which the map inflates obstacle cost values.
+  inflation_radius: 0.55 # default: 0.55
```
Comment (ipa-flg) [inline cob_navigation_config/cob3-6/costmap_common_params.yaml:94]: same as above

File: cob_navigation_config/cob4-2/costmap_common_params.yaml [hunk 3]
```diff
@@ -1,28 +1,89 @@
...
 
-laser_scan_left: {sensor_frame: base_laser_left_link, topic: /base_laser_left/scan, data_type: LaserScan, marking: true, clearing: true}
+#
+# Configure the inflation_layer
+#
+inflation_layer:
 
-laser_scan_right: {sensor_frame: base_laser_right_link, topic: /base_laser_right/scan, data_type: LaserScan, marking: true, clearing: true}
+  # The radius in meters to which the map inflates obstacle cost values.
+  inflation_radius: 0.55 # default: 0.55
```
Comment (ipa-flg) [inline cob_navigation_config/cob4-2/costmap_common_params.yaml:86]: same as above

File: cob_navigation_config/raw3-1/costmap_common_params.yaml [hunk 4]
```diff
@@ -1,30 +1,107 @@
...
+inflation_layer:
 
-softkinetic_right: {sensor_frame: softkinetic_right_link, topic: /softkinetic_right/depth_registered/points, data_type: PointCloud2, marking: true, clearing: true, min_obstacle_height: 0.12, max_obstacle_height: 1.5}
+  # The radius in meters to which the map inflates obstacle cost values.
+  inflation_radius: 0.55 # default: 0.55
 
-cloud_cam3d: {sensor_frame: head_cam3d_link, topic: /voxel_grid/output, data_type: PointCloud2, marking: true, clearing: true, min_obstacle_height: 0.12}
-# END VOXEL STUFF
+  # A scaling factor to apply to cost values during inflation.
+  cost_scaling_factor: 8.01 # default: 10.0
```
Comment (ipa-flg) [inline cob_navigation_config/raw3-1/costmap_common_params.yaml:107]: same as above

File: cob_navigation_global/config/local_costmap_params.yaml [hunk 5]
```diff
@@ -1,12 +1,33 @@
...
+origin_y: 0.0       # default: 0.0
+
+# Whether or not to use a static map (needs to be false if rolling_window = true)
+static_map: false # default: true
+
+# The list of all the costmap layers that will be loaded
+plugins:
+# The obstacle layer tracks the obstacles as read by the sensor data. The ObstacleCostmapPlugin marks and 
+# raytraces obstacles in two dimensions, while the VoxelCostmapPlugin does so in three dimensions. 
+- {name: obstacle_layer, type: "costmap_2d::VoxelLayer"}
```
Comment (ipa-flg) [inline cob_navigation_global/config/local_costmap_params.yaml:30]: do we want to keep it of VoxelLayer type, instead of ObstacleLayer?

**Comment by ipa-flg** [comment]
CR: ok

File: cob_navigation_config/cob3-2/costmap_common_params.yaml (hunk 1 above)
Comment (mgruhler) [inline cob_navigation_config/cob3-2/costmap_common_params.yaml:94]: agreed

File: cob_navigation_global/config/local_costmap_params.yaml (hunk 5 above)
Comment (mgruhler) [inline cob_navigation_global/config/local_costmap_params.yaml:30]: let's tackle this in a seperate PR as well...

**Comment by mgruhler** [comment]
FR: ok

This needs to be tested on hardware though:
- [x] raw3-3
- [x] cob4-1

**Comment by mgruhler** [comment]
Also, see #67 for settings/parameters that need to be reviewed...

**Comment by mgruhler** [comment]
@ipa-fmw @ipa-nhg do you want to test this on cob4-? before a merge?

Once merged, we should definitely tackle #67 quite soon. The current state is adapting the current parameter settings to the new style, but we should actually also adapt some of the params.

**Comment by mgruhler** [comment]
@ipa-fmw @ipa-nhg ping! Do you want to test this? worked fine on raw...

**Comment by HannesBachter** [comment]
tested with 2dnav_ros_tr, 2dnav_ros_dwt and 2dnav_ipa_eband and worked with all of them

**Comment by floweisshardt** [comment]
tested on cob4-1: OK

**Pull request merged**